api/proto/config/cipherinfo.proto

```diff
@@ -0,0 +1,43 @@
+// Copyright(c) 2017-2018 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+option go_package  = "github.com/lf-edge/eve/api/go/config";
+option java_package = "com.zededa.cloud.uservice.proto";
+
+// Cipher information for sensitive data.
+
+// Security Key Exchange Method
+enum KeyExchangeScheme {
+    KEA_NONE = 0;
+    KEA_ECDH = 1;
+}
+
+// Encryption Scheme for Cipher Payload
+enum EncryptionScheme {
+    SA_NONE = 0;
+    SA_AES_256_CFB = 1;
+}
+
+message CipherInfo {
+    string id = 100;
+
+    KeyExchangeScheme keyExchangeScheme = 1;
+    EncryptionScheme encryptionScheme = 2;
+
+    // Initial Value for ECDH computation
+    bytes initialValue = 3;
+
+    // controller certificate
+    bytes controllerCert = 4;
```

Comment:
I assume this is optional; in fact it should probably be separated from the config and be part of the new cert API otherwise the config will get huge.

Reply:
Actually we can drop this AFAIK and rely on the certificates being retrieved/exchanged separately using the new cert APIs.